WORKSPACE

```diff
@@ -0,0 +1,58 @@
+workspace(
+    name = "builder",
+    managed_directories = {
+        "@npm": ["node_modules"],
+    },
+)
+
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+# Fetch rules_nodejs so we can install our npm dependencies
+http_archive(
+    name = "build_bazel_rules_nodejs",
+    sha256 = "1134ec9b7baee008f1d54f0483049a97e53a57cd3913ec9d6db625549c98395a",
+    urls = ["https://github.com/bazelbuild/rules_nodejs/releases/download/3.4.0/rules_nodejs-3.4.0.tar.gz"],
+)
+
+# Check the rules_nodejs version and download npm dependencies
+# Note: bazel (version 2 and after) will check the .bazelversion file so we don't need to
+# assert on that.
+load("@build_bazel_rules_nodejs//:index.bzl", "check_rules_nodejs_version", "node_repositories", "npm_install")
+
+check_rules_nodejs_version(minimum_version_string = "2.2.0")
+
+# Setup the Node.js toolchain
+node_repositories(
+    node_version = "15.0.0",
+    package_json = ["//:package.json"],
+)
+
+npm_install(
+    name = "npm",
+    args = ["--legacy-peer-deps"],  # TODO: package.json have peer dependencies which are contradictory
+    package_json = "//:package.json",
+    package_lock_json = "//:package-lock.json",
+)
+
+# Bellow is an example of how we could run more than one npm install in a project.
+#
+# For now we have decided to have a single node_modules at the root because it centralizes
+# our dependencies and is more inline with mono-repo.
+#
+# TODO(npm): make sure that the package.json in individual packages/*/package.json have
+#       consistent versions from the global one. This will most likely have to be don
+#       by having those files generated from the main package.json.
+#
```

Comment:
This would be nice one day :smile:

Reply:
Right, so the decision to make is:
1) do we want to centrally manage the versions?
2) or do we want to have sub-packages in each folder? 

(see my other comment)